cmd/traffic_manager/traffic_manager.cc

```diff
@@ -680,30 +682,54 @@ main(int argc, const char **argv)
 
   // Setup the API and event sockets
   std::string rundir(RecConfigReadRuntimeDir());
-  std::string apisock(Layout::relative_to(rundir, MGMTAPI_MGMT_SOCKET_NAME));
   std::string eventsock(Layout::relative_to(rundir, MGMTAPI_EVENT_SOCKET_NAME));
 
   mode_t oldmask = umask(0);
   mode_t newmode = api_socket_is_restricted() ? 00700 : 00777;
 
-  int mgmtapiFD         = -1; // FD for the api interface to issue commands
   int eventapiFD        = -1; // FD for the api and clients to handle event callbacks
   char mgmtapiFailMsg[] = "Traffic server management API service Interface Failed to Initialize.";
 
-  mgmtapiFD = bind_unix_domain_socket(apisock.c_str(), newmode);
-  if (mgmtapiFD == -1) {
-    mgmt_log("[WebIntrMain] Unable to set up socket for handling management API calls. API socket path = %s\n", apisock.c_str());
-    lmgmt->alarm_keeper->signalAlarm(MGMT_ALARM_WEB_ERROR, mgmtapiFailMsg);
-  }
-
   eventapiFD = bind_unix_domain_socket(eventsock.c_str(), newmode);
   if (eventapiFD == -1) {
     mgmt_log("[WebIntrMain] Unable to set up so for handling management API event calls. Event Socket path: %s\n",
              eventsock.c_str());
   }
 
   umask(oldmask);
-  ink_thread_create(nullptr, ts_ctrl_main, &mgmtapiFD, 0, 0, nullptr);
+
+  // Setup RPC mechanism.
+  mgmt_server = new RPCServerController(newmode);
+
+  if (mgmt_server->bindSocket(MGMTAPI_MGMT_SOCKET_NAME) == -1) {
+    // mgmt_log("[WebIntrMain] Unable to set up socket for handling management API calls. API socket path = %s\n", apisock.c_str());
+    lmgmt->alarm_keeper->signalAlarm(MGMT_ALARM_WEB_ERROR, mgmtapiFailMsg);
+  }
+
+  mgmt_server->registerControlCallback((RECORD_SET), handle_record_set);
```

Comment:
Why the extra set of parens?

Reply:
Consequence of doing a search and replace when I was removed `OpType`. Removed.